src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -4037,6 +4037,13 @@ public override SyntaxNode SwitchSection(IEnumerable<SyntaxNode> expressions, IE
             return SyntaxFactory.SwitchSection(AsSwitchLabels(expressions), AsStatementList(statements));
         }
 
+        internal override SyntaxNode PatternSwitchSection(IEnumerable<SyntaxNode> labels, IEnumerable<SyntaxNode> statements)
```

Comment:
ah...  noodling on how i feel about this.  An issue for me is that your form really has nothing to do with 'Patterns'.  You're just simply providing a version that takes in Switch-Labels instead of Expressions.  Ideally, you could just pass expressions and labels to the original SwitchSection.  However, the fact thta it calls the parameter 'expressions' is then misleading/annoying.  

Perhaps SwitchSectionFromLabels would be a better name.  As this is internal, i'm ok with that name.  When we go to make these public we can figure out a better name.  However, PatternSwitchSection feels very wrong to me.